Zend/zend_object_handlers.c

```diff
@@ -946,6 +946,18 @@ ZEND_API zval *zend_std_read_property(zend_object *zobj, zend_string *name, int
 				goto exit;
 			}
 
+			if (UNEXPECTED(guard)) {
+				uint32_t guard_type = (type == BP_VAR_IS) && zobj->ce->__isset
+					? IN_ISSET : IN_GET;
+				guard = zend_get_property_guard(zobj, name);
+				if (!((*guard) & guard_type)) {
+					(*guard) |= guard_type;
+					retval = zend_std_read_property(zobj, name, type, cache_slot, rv);
+					(*guard) &= ~guard_type;
+					return retval;
+				}
```

Comment:
I think this can be slightly simplified:

```suggestion
				uint32_t *instance_guard = zend_get_property_guard(zobj, name);
				uint32_t prev_guard = *instance_guard;
				*instance_guard = *guard;
				retval = zend_std_read_property(zobj, name, type, cache_slot, rv);
				*instance_guard = prev_guard;
				return retval;
```

Reply:
Unfortunately this may remove guards from the instance, which may be observable. The instance is an independent object that may be accessed directly, and may have active guards.

Reply:
Ah, I missed that. This looks fine then!